### PR TITLE
Validate email for member models

### PR DESCRIPTION
**Pull request by Zeegaan**
Fixes https://github.com/umbraco/Umbraco-CMS/issues/17512

# Notes 
- Implement string extension method to check whether a string is valid email
- Add check in the ValidateModel of MemberService, to check whether a member is given a valid email.

# How to test
- Follow steps in original issue.

### CODE REVIEW

File: src/Umbraco.Core/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -1558,6 +1559,14 @@ public static IEnumerable<string> EscapedSplit(this string value, char splitChar
         yield return sb.ToString();
     }
 
+    /// <summary>
+    ///     Checks whether a string is a valid email address.
+    /// </summary>
+    /// <param name="email">The string check</param>
+    /// <returns>Returns a bool indicating whether the string is an email address.</returns>
+    public static bool IsEmail(this string email) =>
+        new EmailAddressAttribute().IsValid(email);
```
Comment (bjarnef) [inline src/Umbraco.Core/Extensions/StringExtensions.cs:1568]: @Zeegaan should this do a bit more that using `EmailAddressAttribute`?
https://stackoverflow.com/questions/36947823/validating-an-email-string-in-net-using-emailaddressattribute-but-not-on-an-at

E.g. return `false` when value (email) is null, empty or whitespace?

Comment (Zeegaan) [reply src/Umbraco.Core/Extensions/StringExtensions.cs:1568]: Yea you're probably right, looks like empty or whitespace is already covered, but I've added a null check 👍

File: tests/Umbraco.Tests.UnitTests/Umbraco.Core/ShortStringHelper/StringExtensionsTests.cs [hunk 2]
```diff
@@ -364,6 +364,13 @@ public void IsFullPath()
         TryIsFullPath("   ", false, false); // technically, a valid filename on Linux
     }
 
+    [TestCase("test@test.com",true)]
+    [TestCase("test@test",  true)]
+    [TestCase("testtest.com",  false)]
+    [TestCase("test@test.dk",  true)]
+    [TestCase("test@test.se",  true)]
```
[the remark this answers is not in the document]
Comment (Zeegaan) [reply tests/Umbraco.Tests.UnitTests/Umbraco.Core/ShortStringHelper/StringExtensionsTests.cs:371]: Good point, added 👍

File: src/Umbraco.Infrastructure/Services/MemberEditingService.cs [hunk 3]
```diff
@@ -393,4 +399,6 @@ private async Task<bool> ChangePassword(IMember member, string? oldPassword, str
 
         return true;
     }
+
+    private static bool IsEmailValid(string email) => new EmailAddressAttribute().IsValid(email);
```
Comment (elit0451) [inline src/Umbraco.Infrastructure/Services/MemberEditingService.cs:403]: Is this method needed needed anymore? We can use the new extension method like `if (model.Email.IsEmail() is false)`

**Comment by elit0451** [comment]
Also we need to remember to cherry-pick this to Umbraco 14 ❗

File: src/Umbraco.Infrastructure/Services/MemberEditingService.cs (hunk 3 above)
Comment (Zeegaan) [reply src/Umbraco.Infrastructure/Services/MemberEditingService.cs:403]: Nope it's not needed, good catch! 💪 
Removed 🚀

**Review by elit0451 — APPROVED** [review]
Looks good 💪

**Pull request merged**